analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -219,6 +215,16 @@ private static SymbolicConstraint BinaryConstraint(BinaryOperatorKind kind, Symb
         }
     }
 
+    private static BoolConstraint BinaryNullableBoolConstraint(BinaryOperatorKind kind, BoolConstraint boolConstraint, bool oneIsNull) =>
```

Comment:
`oneIsNull` makes sense in scope of this refactoring, but will be hard to understand without the refactoring context.
I also wouldn't mind passing both values in, as `when leftIsNull || rightIsNull` is very clear to understand
```suggestion
    private static BoolConstraint BinaryNullableBoolConstraint(BinaryOperatorKind kind, BoolConstraint boolConstraint, bool hasNull) =>
```